implementations/rust/ockam/src/queue.rs

```diff
@@ -0,0 +1,44 @@
+use crate::address::Addressable;
+use alloc::collections::VecDeque;
+
+pub trait Queue<T> {
+    fn enqueue(&mut self, element: T) -> crate::Result<bool>;
+    fn dequeue(&mut self) -> Option<T>;
+    fn is_empty(&self) -> bool;
+}
```

Comment:
I think all three of us (you, Jared and I) have all implemented some version of a queue!

Reply:
this code is in this PR only because my branch is based on Jared's queue PR  #748

Reply:
so one less queue 😄